File: programs/rand.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <stdint.h>
#include <inttypes.h>
#include <time.h>
#define BOARDSIZE 15
#define EMPTY 0
#define BLACK 1
#define WHITE 2

typedef struct { int8_t x, y; } Position;
typedef struct { int8_t conti; bool empty; } Peeked;

Position run(int8_t board[], int8_t stone, double time);
void parse(int8_t board[], int8_t* stone, double* time);

Position run(int8_t board[], int8_t stone, double time) {
    Position pos = { 0, 0 };

    while (board[pos.y * BOARDSIZE + pos.x] != EMPTY) {
        pos.x = rand() % BOARDSIZE;
        pos.y = rand() % BOARDSIZE;
    }

    return pos;
}

int main() {
    srand(time(NULL));

    int8_t* board = calloc(225, sizeof(int8_t));
    int8_t stone = 0;
    double time = 0;

    parse(board, &stone, &time);

    Position selected = run(board, stone, time);

    printf("%" PRId8 " %" PRId8 "\n", selected.y, selected.x);

    return EXIT_SUCCESS;
}

void parse(int8_t board[], int8_t* stone, double* time) {
    for (size_t i = 0; i < 225; i++) {
        scanf("%" SCNd8 ", ", board + i);
    }
    scanf("%" SCNd8 ", %lf", stone, time);
}

File: programs/rand.go
package main

import (
	"fmt"
	"math/rand"
)

var (
	board [][]int32
	stone int32
	time  float64
)

func main() {
	parse()

	var x, y = run()

	fmt.Printf("%d, %d\n", x, y)
}

func run() (int32, int32) {
	var x, y = int32(0), int32(0)

	for board[x][y] != 0 {
		x, y = rand.Int31n(15), rand.Int31n(15)
	}

	return x, y
}

func parse() {
	for i := 0; i < 15; i++ {
		board = append(board, []int32{})
		for j := 0; j < 15; j++ {
			board[i] = append(board[i], 0)
			fmt.Scanf("%d,", &board[i][j])
		}
	}

	fmt.Scanf("%d, %f", &stone, &time)

	// fmt.Fprintln(os.Stderr, board, stone, time)
}
